wfl/fit/ace.py

```diff
@@ -138,50 +208,61 @@ def _check_output_files(file_base, formats):
 
         return results
 
-    if not run_dir.exists():
-        run_dir.mkdir(parents=True)
+    run_dir.mkdir(exist_ok=True, parents=True)
 
-    use_params = dict(params)
-
-    if dry_run:
-        use_params['dry_run'] = None
-
-    # the code below needs to know an unfortunate amount about the inner workings of ace_fit.jl
-    # here we rely on knowledge of the --outfile_base
-    use_params['outfile_base'] = str(ace_file_base)
-    use_params['outfile_format'] = formats
-
-    use_params['key'] = [ [k, ref_property_prefix + v] for k,v in [('E', 'energy'), ('F', 'forces'), ('V', 'virial')] ]
-    use_params['_repeat_key'] = True
+    use_params = deepcopy(ace_fit_params)
 
     # configs need to be in memory so they can be modified with stress -> virial, and safest to
     # have them as a list (rather than using ConfigSet_in.to_memory()) when passing to ase.io.write below
     fitting_configs = list(fitting_configs)
 
-    # calculate virial from stress, since ASE uses stress but ace_fit.jl only knows about virial
-    for at in fitting_configs:
-        if ref_property_prefix + 'stress' in at.info:
-            stress = at.info[ref_property_prefix + 'stress']
-            if len(stress.shape) == 2:
-                # 3x3
-                at.info[ref_property_prefix + 'virial'] = list((-stress * at.get_volume()).ravel())
-            else:
-                # Voigt 6-vector
-                at.info[ref_property_prefix + 'virial'] = list((-voigt_6_to_full_3x3_stress(stress) * at.get_volume()).ravel())
+    fit_cfgs_fname_given = None 
```

Comment:
Workflow paths override whatever was in the ace parameters given to these functions. That only applies to `ACE_fname` and `*params["data"]["fname"]`. In both cases, a warning is issued if something is actually being overwritten. 

`*params["data"]["fname"]` is mandatory in ACE1pack, but not in Workflow, because `fit()` takes configs to fit to and writes them to file itself. So there's some handling that in case some of the fields aren't present in the dictionary.

Reply:
That's what I figured, but it's still hard to figure out what's going to happen, with the parameters dict and the explicit function argument both being possible ways it can be specified.  That's why I want a comment explaining it.

Reply:
Of course you meant the code. Have separated that out as a function for clarity and made the note.